Fix the front end module permissions

aschempp opened the pull request:
I have found several issues in the current frontend module permissions.

1. it did not check access to the module itself
2. I could not find that `USER_CAN_DELETE_FRONTEND_MODULE` would be for, I don't think that actually worked
3. Voters should not vote on the user object themselves, they all currently forward to the `BackendPermissionVoter`

/cc @bezin

aschempp committed:
Fixed frontend module permissions

aschempp committed:
Fixed tests

aschempp commented:
I also noticed that the permission voter currently allows access to all modules if no module permissions are given. This is contrary to how any other permissions work, and does not really make much sense because we also have a migration that enables all front end modules for all existing user groups 🤔

aschempp committed:
Added missing internal flag

aschempp commented:
~Next issue found:~
 1. ~should a user be able to select a module in the module content element, if no access to that module was given (i.e. see it in the dropdown)?~
 2. ~if yes (and that's the current state), it always renders an edit button, but that will result in an error if access to that module is not given.~

Nevermind, it seems to work in the online demo.

bezin commented:
> I also noticed that the permission voter currently allows access to all modules if no module permissions are given. This is contrary to how any other permissions work, and does not really make much sense because we also have a migration that enables all front end modules for all existing user groups 🤔

Yes, I agree. My intention was to imitate the behaviour of content element permissions, so I checked how it was implemented there, but I was mistaken, I guess. 

Front-end modules must be explicitly allowed, otherwise no access is permitted (like content elements).

aschempp committed:
Check module type access the same way we check for page types

leofeyer reviewed (changes requested):
Something seems to be wrong, because now a regular user can delete all front end modules and not just the types that are allowed for them.

aschempp committed:
Fix the class registration

aschempp committed:
Disable the pencil icon if user does not have access to modules

aschempp commented:
> Something seems to be wrong, because now a regular user can delete all front end modules and not just the types that are allowed for them.

Fixed, the service was not correctly registered because I renamed the class 🙈

leofeyer committed:
CS

leofeyer commented:
Thank you @aschempp.
